tests/gem5/chi_protocol/test_chi_per_isa.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2024 The Regents of the University of California
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""
+
+"""
+
+from testlib import (
+    absdirpath,
+    config,
+    constants,
+    joinpath,
+    verifier,
+)
+
+from gem5.suite import gem5_verify_config
+
+if config.bin_path:
+    resource_directory = config.bin_path
+else:
+    resource_directory = joinpath(
+        config.base_dir, "tests", "gem5", "resources"
+    )
+
+for isa in ("arm", "riscv", "x86"):
+    for num_cores in (1, 2, 4):
+        for atomic in (True, False):
```

Comment:
We should only test timing. As Hoa said, Ruby doesn't work with atomic.

Reply:
I just wanted a test to check it wasn't catastrophically incompatible. I.e., things still run, the simulation didn't crash. Though i'm aware ruby doesn't work with atomic you can still put ruby caches in simulations that are run with atomic cores and doing so shouldn't cause issue. These tests are very fast, so the cost covering the cross-product of possible setups isn't costly. Regardless, the tests may send the wrong message to users (I suppose someone may thing "if they are testing it, it must be a valid config, right?") so I'll remove it.

(tl;dr: ok)